timesketch/frontend-ng/src/components/Explore/AggregateDialog.vue

```diff
@@ -0,0 +1,622 @@
+          },
+          categories: categories
+        },
+      }
+    },
+    intervalChartSeries() {
+      if (this.stats === undefined) return []
+
+      let data = []
+
+      switch (this.selectedDistributionInterval) {
+        case "Year":
+          data = Array(this.stats.year_histogram.length).fill(0)
+          for (const [index, entry] of this.stats.year_histogram.buckets.entries()) {
+            data[index] = entry.doc_count
+          }
+          break
+        case "Month":
+          data = Array(12).fill(0)
+          for (const entry of this.stats.month_histogram.buckets) {
+            data[entry.key - 1] = entry.doc_count
+          }
+          break
+        case "Day":
+          data = Array(7).fill(0)
+          for (const entry of this.stats.day_histogram.buckets) {
+            data[entry.key - 1] = entry.doc_count
+          }
+          break
+        case "Hour":
+          data = Array(24).fill(0)
+          for (const entry of this.stats.hour_histogram.buckets) {
+            data[entry.key] = entry.doc_count
+          }
+          break
+        case "Hour-Day":
+          return this.intervalHeatmapSeries
+        default:
+          break
+      }
+      return [
+        {
+          name: 'Events by ' + this.selectedDistributionInterval,
+          data: data
+        }
+      ]
+    },
+    fieldCardinality() {
+      if (this.stats === undefined) return null
+      return this.stats.all_values.field_cardinality.value
+    },
+    docCount() {
+      if (this.stats === undefined) return null
+      return this.stats.all_values.doc_count
+    },
+    fieldDateTimeMinimum() {
+      if (this.stats === undefined) return null
+      return this.stats.all_values.datetime_min.value_as_string
+    },
+    fieldDateTimeMaximum() {
+      if (this.stats === undefined) return null
+      return this.stats.all_values.datetime_max.value_as_string
+    },
+    valueDateTimeMinimum() {
+      if (this.stats === undefined) return null
+      return this.stats.datetime_min.value_as_string
+    },
+    valueDateTimeMaximum() {
+      if (this.stats === undefined) return null
+      return this.stats.datetime_max.value_as_string
+    },
+    fieldValueCount() {
+      if (this.stats === undefined) return null
+      return this.stats.all_values.field_count.value
+    },
+    valueEventCount() {
+      if (this.stats === undefined) return null
+      return this.stats.value_count.value
+    },
+    commonValues() {
+      if (this.stats === undefined) return []
+      return this.stats.all_values.top_terms.buckets
+    },
+    rareValues() {
+      if (this.stats === undefined) return []
+      return this.stats.all_values.rare_terms.buckets
+    },
+    donutChartOptions() {
+      return {
+        chart: {
+          type: 'donut'
+        },
+        dataLabels: {
+          enabled: true
+        },
+        labels: [
+          this.eventValue + ' (' + this.valueEventCount + ')',
```

Comment:
```suggestion
          this.truncateValue(this.eventValue)  + ' (' + this.valueEventCount + ')',
```
* The `truncateValue()` function can also be applied to labels to make them more readable.